src/python/WMComponent/PhEDExInjector/PhEDExInjectorPoller.py

```diff
@@ -36,17 +63,35 @@ class PhEDExInjectorPoller(BaseWorkerThread):
 
     Poll the DBSBuffer database and inject files as they are created.
     """
-    def __init__(self, config, phedex, nodeMappings):
+    def __init__(self, config):
         """
         ___init___
 
         Initialise class members
         """
         BaseWorkerThread.__init__(self)
-        self.config = config
-        self.phedex = phedex
         self.dbsUrl = config.DBSInterface.globalDBSUrl
 
+        self.pollCounter = 0
+        self.subFrequency = None
+        if getattr(config.PhEDExInjector, "subscribeDatasets", False):
+            pollInterval = config.PhEDExInjector.pollInterval
+            subInterval = config.PhEDExInjector.subscribeInterval
+            self.subFrequency = max(1, int(round(subInterval/pollInterval)))
+            logging.info("SubscribeDataset and deleteBlocks will run every %d polling cycles", self.subFrequency)
+
+        # retrieving the node mappings is fickle and can fail quite often
+        self.phedex = PhEDEx({"endpoint": config.PhEDExInjector.phedexurl}, "json")
+        try:
```

Comment:
We really should get the retry package as part of our distribution to get rid of silly patterns like this. @ticoann ?